server/src/main/java/org/opensearch/search/startree/filter/provider/StarDateFieldMapper.java

```diff
@@ -0,0 +1,152 @@
+import org.opensearch.index.compositeindex.datacube.startree.utils.date.DateTimeUnitRounding;
+import org.opensearch.index.mapper.CompositeDataCubeFieldType;
+import org.opensearch.index.mapper.DateFieldMapper.DateFieldType;
+import org.opensearch.index.mapper.MappedFieldType;
+import org.opensearch.index.query.RangeQueryBuilder;
+import org.opensearch.search.internal.SearchContext;
+import org.opensearch.search.startree.StarTreeQueryHelper;
+import org.opensearch.search.startree.filter.DimensionFilter;
+import org.opensearch.search.startree.filter.RangeMatchDimFilter;
+
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.LongSupplier;
+
+class StarDateFieldMapper implements DimensionFilterMapper {
+
+    CompositeDataCubeFieldType compositeDataCubeFieldType;
+    LongSupplier nowSupplier;
+    String subDimensionField;
+
+    public StarDateFieldMapper(SearchContext searchContext) {
+        this.nowSupplier = () -> searchContext.getQueryShardContext().nowInMillis();
+        this.compositeDataCubeFieldType = (CompositeDataCubeFieldType) searchContext.mapperService()
+            .getCompositeFieldTypes()
+            .iterator()
+            .next();
+    }
+
+    @Override
+    public DimensionFilter getExactMatchFilter(MappedFieldType mappedFieldType, List<Object> rawValues) {
+        // TODO: implement term/terms query on date field to be resolved with star-tree
+        return null;
+    }
+
+    @Override
+    public DimensionFilter getRangeMatchFilter(MappedFieldType mappedFieldType, RangeQueryBuilder rangeQueryBuilder) {
+
+        DateFieldType dateFieldType = (DateFieldType) mappedFieldType;
+        String field = rangeQueryBuilder.fieldName();
+        DateDimension dateDimension = (DateDimension) StarTreeQueryHelper.getMatchingDimensionOrNull(
+            field,
+            compositeDataCubeFieldType.getDimensions()
+        );
+
+        // Convert format string to DateMathParser if provided
+        DateMathParser forcedDateParser = rangeQueryBuilder.format() != null
+            ? DateFormatter.forPattern(rangeQueryBuilder.format()).toDateMathParser()
+            : org.opensearch.index.mapper.DateFieldMapper.getDefaultDateTimeFormatter().toDateMathParser();
+
+        ZoneId timeZone = rangeQueryBuilder.timeZone() != null ? ZoneId.of(rangeQueryBuilder.timeZone()) : null;
+
+        long l = Long.MIN_VALUE;
+        long u = Long.MAX_VALUE;
+
+        if (rangeQueryBuilder.from() != null) {
+            l = DateFieldType.parseToLong(
+                rangeQueryBuilder.from(),
+                !rangeQueryBuilder.includeLower(),
+                timeZone,
+                forcedDateParser,
+                nowSupplier,
+                dateFieldType.resolution()
+            );
+            if (!rangeQueryBuilder.includeLower()) {
+                ++l;
+            }
+        }
+
+        if (rangeQueryBuilder.to() != null) {
```

Comment:
Can we kindly add tests which includes boundaries.

For example documents with :
1:00:00
and 
2:00:00

And running queries with lt,lte,gt,gte combinations

Reply:
This is already added in tests.

Reply:
This is not handled properly as explained with an example - lets add proper tests.

LTE, GTE with star tree ends up with incorrect results.